packages/node/test/machine/integration/protocols/install-uninstall.spec.ts

```diff
@@ -0,0 +1,103 @@
+import { OutcomeType } from "@counterfactual/types";
+import { Two, Zero } from "ethers/constants";
+
+import { CONVENTION_FOR_ETH_TOKEN_ADDRESS } from "../../../../src/constants";
+import { toBeEq } from "../bignumber-jest-matcher";
+
+import { Participant, TestRunner } from "./test-runner";
+
+expect.extend({ toBeEq });
+
+export enum TokenType {
+  ETH = "ETH",
+  ERC20 = "ERC20",
+  SPLIT = "SPLIT"
+}
+
+async function runDirectInstallUninstallTest(
+  outcomeType: OutcomeType,
+  tokenType: TokenType
+) {
+  const tr = new TestRunner();
+  await tr.connectToGanache();
+
+  await tr.setup();
+  await tr.unsafeFund();
+
+  if (tokenType === TokenType.SPLIT) {
+    await tr.installSplitDeposits(
+      outcomeType,
+      CONVENTION_FOR_ETH_TOKEN_ADDRESS,
+      TestRunner.TEST_TOKEN_ADDRESS
+    );
+    tr.assertFB(Participant.A, CONVENTION_FOR_ETH_TOKEN_ADDRESS, Zero);
+    tr.assertFB(Participant.B, TestRunner.TEST_TOKEN_ADDRESS, Zero);
+
+    await tr.uninstall();
+    tr.assertFB(Participant.A, CONVENTION_FOR_ETH_TOKEN_ADDRESS, Two);
+    tr.assertFB(Participant.B, TestRunner.TEST_TOKEN_ADDRESS, Zero);
+  } else {
+    const tokenAddress = {
+      [TokenType.ETH]: CONVENTION_FOR_ETH_TOKEN_ADDRESS,
+      [TokenType.ERC20]: TestRunner.TEST_TOKEN_ADDRESS
+    }[tokenType];
+
+    await tr.installEqualDeposits(outcomeType, tokenAddress);
+    tr.assertFB(Participant.A, tokenAddress, Zero);
+    tr.assertFB(Participant.B, tokenAddress, Zero);
+    await tr.uninstall();
+    tr.assertFB(Participant.A, tokenAddress, Two);
+    tr.assertFB(Participant.B, tokenAddress, Zero);
+  }
+}
+
+describe("Install-then-uninstall in a direct channel", () => {
+  for (const outcomeType of [
```

Comment:
I think this entire for lop would be better off written  as explicit it statements

Reply:
I disagree, it's much more LOC

Reply:
but less readable -- I prefer tests that are readable

Reply:
I don't think it will be more readable - the main worry is that you would have to scrutinize the 7 `it` statements, mentally reconstruct the 3x3 matrix of which features are tested (and which of those combinations are disabled), and then make sure that the 7 `it` statements actually implement the matrix properly